app/frontend/chat_interface.py

```diff
@@ -462,4 +465,10 @@ def get_dynamic_chat_placeholder():
             st.rerun()
 
     if hasattr(user_input, 'files') and user_input.files:
-        st.image(user_input.files[0], width=400)
\ No newline at end of file
+        img = Image.open(user_input.files[0])
+        width, height = img.size
```

Comment:
The user-uploaded file is opened without closing, which may lead to resource leaks. Use a context manager when opening files.
```
with Image.open(user_input.files[0]) as img:
    width, height = img.size
```